src/graph_gui.cpp

```diff
@@ -216,6 +217,14 @@ struct BaseGraphWindow : Window {
 	uint8_t highlight_range = UINT8_MAX; ///< Data range that should be highlighted, or UINT8_MAX for none.
 	bool highlight_state = false; ///< Current state of highlight, toggled every TIMER_BLINK_INTERVAL period.
 
+	template <typename Tprojection>
+	struct Filler {
+		DataSet &dataset; ///< Dataset to fill.
+		const Tprojection &proj; ///< Projection to apply.
+
+		inline void Fill(uint i, const auto &data) const { this->dataset.values[i] = std::invoke(this->proj, data); }
```

Comment:
Why `uint` instead of `size_t`?

Reply:
`size_t` felt unnecessary given the range. But then so is `uint`, I guess.